pkg/pot/lex.go

```diff
@@ -215,169 +99,140 @@ func Lex(r io.Reader) ([]Token, error) {
-func parseMsgString(line string) (string, error) {
-	if !hasValidPrefix(line) {
-		return "", errors.New("incorrect format of po tags")
-	}
+	return &token, nil
+}
 
+// parseValue parses the value of the token.
+// e.g.
+//
+//	msgid "some text" -> some text
+//	#, python-format -> python-format
+//	#: superset/key_value/exceptions.py:54 -> superset/key_value/exceptions.py:54
+func parseValue(line string) string {
 	n := 2
-	fields := strings.SplitN(line, " ", n)
-
-	var tokenValue string
+	fields := strings.SplitN(line, " ", n) // prefix and value, e.g. fields[0] = msgid, fields[1] = "text", etc.
 
-	if len(fields) == n {
-		tokenValue = strings.TrimSpace(fields[1])
+	// No value, only prefix
+	if len(fields) != n {
+		return ""
 	}
 
-	if strings.HasPrefix(tokenValue, `"`) && strings.HasSuffix(tokenValue, `"`) {
-		// Remove the quotes and any escaped quotes
-		tokenValue = strings.ReplaceAll(tokenValue[1:len(tokenValue)-1], `\"`, `"`)
-	} else if strings.HasSuffix(tokenValue, "\\n\"") {
-		tokenValue = strings.ReplaceAll(tokenValue[:len(tokenValue)-2], "\\", ``)
-		tokenValue = strings.TrimSpace(tokenValue)
-	}
-
-	return tokenValue, nil
+	return trimQuotes(fields[1])
 }
 
-// parseMultilineToken function modifies the last token in the slice
-// by appending a parsed multiline string from the line. The modified token's value is trimmed of whitespace,
-// and the function returns nil for both the token pointer and the error.
-func parseMultilineToken(line string, tokens *[]Token) (*Token, error) {
-	lastToken := (*tokens)[len(*tokens)-1]
+// parseMultilineValue parses the value of the multiline msgid, msgstr, msgid_plural, msgstr[*] tokens.
+// Instead of returning a parsed value, it appends it to the last token in the tokens slice.
+func parseMultilineValue(line string, tokens *[]Token) error {
+	lastToken := &(*tokens)[len(*tokens)-1]
+
 	switch lastToken.Type { //nolint:exhaustive
 	case TokenTypeMsgID, TokenTypePluralMsgID, TokenTypeMsgStr, TokenTypePluralMsgStr:
-		lastToken.Value += " " + parseMultilineString(line)
-		lastToken.Value = strings.TrimSpace(lastToken.Value)
-		(*tokens)[len(*tokens)-1] = lastToken
+		lastToken.Value += "\n" + trimQuotes(line)
+	default:
+		return fmt.Errorf("unsupported multiline string for token type: '%d'", lastToken.Type)
 	}
 
-	return nil, nil //nolint:nilnil
-}
-
-// parseMultilineString removes all double quote characters from the input line string and returns the modified string.
-func parseMultilineString(line string) string {
-	return strings.ReplaceAll(line, `"`, "")
+	return nil
 }
 
-// hasValidPrefix determines whether the provided line string has a valid prefix
-// by iterating through a collection of valid prefixes.
-func hasValidPrefix(line string) bool {
-	for _, prefix := range validPrefixes {
-		if strings.HasPrefix(line, prefix) {
-			return true
-		}
-	}
+// trimQuotes removes first and last double quotes from the string, if they exist.
```

Comment:
No need to create a helper func. Simply use directly `strings.Trim(v, `"`).

PS. There is a side effect that I don't like - quotes are not optional when parsing. They are required.

Reply:
I mean, we should error if there are no quotes present.

Reply:
they are required in msgid, msgstr, and in headers
but for example in comments they are not.

Also they have differences

as in 


```
"Language: en-US\n" // prefix: Language: Value: en-US\n"
msgid "some text"   // prefix msgid: Value: "some-text"
#, python-format    // prefix:  # Value: python-format
```

Reply:
We have to handle it correctly.

Reply:
reverted to simple strings.Trim()

reason: Too many scenarios to handle.  Also case where file is formatted correctly is more common, than incorrectly (Pareto). So I just go with that, to save time